archeryutils/handicaps/handicap_equations.py

```diff
@@ -355,15 +356,126 @@ def sigma_r(
     return sig_r
 
 
+def s_bar(
+        target_specs: dict[float, int],
+        arw_rad: float,
+        sig_r: FloatArray
+    ) -> FloatArray:
+    """Calculate expected score directly from target ring sizes.
+
+    Parameters
+    ----------
+    target_specs : dict[float, int]
+        Mapping of target ring *diameters* in [metres], to points scored
```

Comment:
Interesting question here would be if we want to include `X` as an option?
I can see this 'target spec' representation being useful for things beyond handicap code...

Reply:
Also worth discussing for the purposes of a constructor if we want to use diameter in m or cm, and how to avoid a potential conflict with the currently supplied `diameter`.

One option would be to provide the fraction of the face for each ring, but this is less intuitive and gets messy for non-full faces.
Another option is to raise a warning to say `diameter` is being ignored/overridden if specifying a `"custom"` scoring setup, but this could get irritating appearing every time (`diameter` is currently a required variable) and providing a diameter and spec is unintuitive.
Another alternative is to check to enforce diameter and spec match, but this is superfluous and could be confusing for non-full faces.

I think perhaps the answer is to keep things largely as they are now, but implement a `@classmethod` called `custom_scoring` or similar that takes a `spec` and returns a target using it. See [here](https://realpython.com/python-multiple-constructors/#providing-multiple-constructors-with-classmethod-in-python) for an example.

Thoughts?

Reply:
Looked a bit more and can do the same trick we've done with diameter and distance units. The constructor can accept either a specification dictionary in which case metres is assumed, or a tuple of specification and unit string in which case the keys are converted into metres before storing.

Question for you on the docstring for the alternate constructor; given several of the parameters are the same as for the normal class constructor and are just passed through, is it still best to duplicate the information in the docstring and then update it twice if it changes, or just refer back to the other docstring?

Reply:
Mmmmm good question.
I think for completeness it makes sense to duplicate things for the purposes of documentation, even if it feels a bit verbose.
One would hope that as a code functionality those parts are unlikely to change.

At the risk of flip-flopping, given `diameter` is still required I wonder could it be that we make `face_spec` an optional argument to the original constructor that triggers other logic? This has potential to bring conflict with the `system` argument however. After some reflection I think having a simple basic constructor and an independent 'special' constructor for custom targets is the way to go - rather than trying to make an over-complicated catch-all super-constructor.